python/caches/tests/test_hdf_cache.py

```diff
@@ -0,0 +1,21 @@
+import pytest
+from hydrotools.caches.hdf import HDFCache
+import pandas as pd
+import numpy as np
+from pathlib import Path
+
+cache_path = Path('test_cache.h5')
```

Comment:
For testing purposes, it might be best if we move over to using `NamedTemporaryFile`s. It would probably be best to either return the name of a temporary file from a fixture or do it within the context of a test itself.

Other tests that would be nice to see:
- using a byte string as a key.
- using different string encodings as keys of the same name
- Getting a `df` from the cache outside of the context of a context manager (from an instance of `HDFCache`).

Reply:
I used a `TemporaryDirectory`. `tempfile` returns file buffers and `HDFStore` needs a path string.

I added a couple more tests and now raise for non-string keys.

Reply:
Okay great! Thanks for doing that.